Rebuild for python 3.14 free-threading

beenje opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

coverage is supporting free-threading. There are wheels available on pypi for 3.14t.
Apply the 3.14t migration manually.

I had issues with conda-build which was mixing python 3.14 and 3.14t.
I converted the recipe to v1 format.
`track_features` isn't available in the new format. I used `down_prioritize_variant` instead.

beenje commented:
When building for 3.14t on linux 64 there seems to be some confusion. It's using 3.14 instead of 3.14t...

```
2025-10-30T12:38:30.9527944Z     pip:              25.2-pyh145f28c_0         conda-forge
2025-10-30T12:38:30.9528215Z     python:           3.14.0-h32b2ec7_102_cp314 conda-forge
2025-10-30T12:38:30.9528504Z     python_abi:       3.14-8_cp314              conda-forge

2025-10-30T12:38:43.4110600Z Using pip 25.2 from $PREFIX/lib/python3.14/site-packages/pip (python 3.14)
```

Update 2025-11-18: This was an issue with conda-build. Fixed by moving to rattler-build.

beenje committed:
Convert recipe to v1 format

beenje committed:
Convert recipe to v1 format

beenje committed:
Bump build number

beenje committed:
Bump build number

beenje committed:
Add python 3.14t migration

beenje committed:
Add python 3.14t migration

beenje committed:
MNT: Re-rendered with conda-smithy 3.53.1 and conda-forge-pinning 2025.11.18.16.36.32

Other tools:
- conda-build 25.9.0
- rattler-build 0.50.0
- rattler-build-conda-compat 1.4.9

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipe/recipe.yaml**:

 * ❌ Invalid `noarch` value `${{ "python" if use_noarch }}`. Should be one of `python`, `generic`.
 * ❌ `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: ${{ "python" if use_noarch }}`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/19475868850](https://github.com/conda-forge/conda-forge-webservices/actions/runs/19475868850). Examine the logs at this URL for more detail.</sub>

beenje commented:
> For **recipe/recipe.yaml**:
> 
> * ❌ Invalid `noarch` value `${{ "python" if use_noarch }}`. Should be one of `python`, `generic`.
> * ❌ `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: ${{ "python" if use_noarch }}`.
> 

This feature was added in `rattler-build` 0.50.0: https://github.com/prefix-dev/rattler-build/pull/1973
It's working fine as seen in the build logs.

I guess the linter has to be updated?

beenje commented:
@conda-forge/coverage this PR is ready.

Does the linter need to be fixed before it can be merged?

beenje committed:
Build osx-arm64 natively

beenje committed:
MNT: Re-rendered with conda-smithy 2026.8.9 and conda-forge-pinning 2026.08.09.00.32.18

Other tools:
- conda-build 26.7.0
- rattler-build 0.72.2
- rattler-build-conda-compat 1.4.19
